src/settings/DeviceTokenScreen.js

```diff
@@ -0,0 +1,137 @@
+/* @flow strict-local */
+
+import * as React from 'react';
+import { Platform } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import Clipboard from '@react-native-clipboard/clipboard';
+
+import type { RouteProp } from '../react-navigation';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+import Screen from '../common/Screen';
+import { createStyleSheet } from '../styles';
+import { useGlobalSelector, useSelector } from '../react-redux';
+import { getGlobalSession } from '../directSelectors';
+import { getAccount } from '../account/accountsSelectors';
+import { showToast } from '../utils/info';
+import { TranslationContext } from '../boot/TranslationProvider';
+import ZulipText from '../common/ZulipText';
+import AlertItem from '../common/AlertItem';
+
+type Props = $ReadOnly<{|
+  navigation: AppNavigationProp<'device-token'>,
+  route: RouteProp<'device-token', void>,
+|}>;
+
+export default function DeviceTokenScreen(props: Props): React.Node {
+  const _ = React.useContext(TranslationContext);
+
+  const pushToken = useGlobalSelector(state => getGlobalSession(state).pushToken);
+  const account = useSelector(getAccount);
+  const { ackedPushToken, realm } = account;
+
+  const handlePressCopyDeviceToken = React.useCallback(() => {
+    if (pushToken == null) {
+      // This shouldn't happen
+      return;
+    }
+
+    Clipboard.setString(pushToken);
+    showToast(_('Device token copied'));
+  }, [pushToken, _]);
+
+  const handlePressCopyAckedToken = React.useCallback(() => {
+    if (ackedPushToken == null) {
+      // This shouldn't happen
+      return;
+    }
+
+    Clipboard.setString(ackedPushToken);
+    showToast(_('Registered token copied'));
+  }, [ackedPushToken, _]);
+
+  const styles = React.useMemo(
+    () =>
+      createStyleSheet({
+        tokenText: {
+          fontFamily:
+            Platform.OS === 'ios'
+              ? 'Courier' // This doesn't seem to work on Android
+              : 'monospace', // This doesn't seem to work on iOS
```

Comment:
Hilarious.

Instead of Courier, perhaps "SF Mono"? That's Apple's system monospace font:
https://developer.apple.com/fonts/

Reply:
Hmm, I get "Unrecognized font family 'SF Mono'".

Reply:
I chose the `fontFamily` from what worked in src/RootErrorBoundary.js:

```js
          {Platform.select({
            ios: (
              <TextInput
                style={{
                  // To not grow outside the parent View
                  flex: 1,

                  // This doesn't seem to work on Android
                  fontFamily: 'Courier',
                }}
                multiline
                editable={false}
                value={details}
              />
            ),
            android: (
              // For Android-only symptoms like
              // facebook/react-native#23117
              <ScrollView>
                <TextInput
                  style={{
                    // To not grow outside the parent View
                    flex: 1,

                    // This doesn't seem to work on iOS
                    fontFamily: 'monospace',
                  }}
                  multiline
                  editable={false}
                  value={details}
                />
              </ScrollView>
            ),
          })}
```